File: 0024-swap-nodes-in-pairs/0024-swap-nodes-in-pairs.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var swapPairs = function(head) {
    let dummy = new ListNode(-1);
    dummy.next = head;
    let prev = dummy;
    
    while (head && head.next) {
        let first_node = head;
        let second_node = head.next;

        // Swap the nodes
        prev.next = second_node;
        first_node.next = second_node.next;
        second_node.next = first_node;

        // Move to the next pair
        prev = first_node;
        head = first_node.next;
    }

    return dummy.next;
};

File: 0024-swap-nodes-in-pairs/0024-swap-nodes-in-pairs.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func swapPairs(head *ListNode) *ListNode {
    dummy := &ListNode{-1, head}
    prev := dummy

    for head != nil && head.Next != nil {
        first_node := head
        second_node := head.Next

        // Swap the nodes
        prev.Next = second_node
        first_node.Next = second_node.Next
        second_node.Next = first_node

        // Move to the next pair
        prev = first_node
        head = first_node.Next
    }

    return dummy.Next
}